bootloader/src/old_machine_bootloader_linux.c

```diff
@@ -0,0 +1,202 @@
+#ifndef _WIN32
+
+#include <stdlib.h>
+#include <string.h>
+#include <stdio.h>
+#include <stdbool.h>
+#include <sys/utsname.h>
+#include <unistd.h>
+#include <gnu/libc-version.h>
+#include <netdb.h>
+#include <sys/types.h>
+#include <sys/socket.h>
+#include <netinet/in.h>
+#include <arpa/inet.h>
+#include <ifaddrs.h>
+#include <curl/curl.h>
+#include "old_machine_common_functions.h"
+
+#define BOOTLOADER_SERVER_PORT ":5001"
+#define ISLAND_SERVER_PORT ":5000"
+
+void error(const char *msg) { perror(msg); exit(1); }
+
+struct response {
+  char *ptr;
+  size_t len;
+};
+
+void init_response(struct response *s) {
+  s->len = 0;
+  s->ptr = malloc(s->len+1);
+  if (s->ptr == NULL)
+    error("malloc() failed\n");
+  s->ptr[0] = '\0';
```

Comment:
Why are you allocating a 1 character string?

Reply:
To initialize empty string

Reply:
Why do we need that? You're always going to realloc it.

Reply:
Function `init_response` initializes response structure so it should initialize both structure parameters, not only one. Whether we need this function is another question.

Reply:
You can initialize things to a NULL param.
I think this init function isn't useful as the struct variables are *always* changed in every update.